diff -rupN /home/pako/Downloads/lstrip/lstrip.c lua/lstrip/lstrip.c
--- /home/pako/Downloads/lstrip/lstrip.c	2012-10-08 15:50:57.000000000 +0200
+++ lua/lstrip/lstrip.c	2014-11-17 10:29:54.595084082 +0100
@@ -32,9 +32,10 @@ static void strip(lua_State *L, const ch
 
 int main(int argc, char* argv[])
 {
- int preserve=0;			/* preserve line breaks? */
+  int preserve=0;			/* preserve line breaks? */
  int dump=0;				/* dump instead of stripping? */
  lua_State *L=luaL_newstate();
+ (void)(argc); // remove compiler warning
  if (argv[0]!=NULL && *argv[0]!=0) progname=argv[0];
  if (L==NULL) fatal("not enough memory for state");
  while (*++argv!=NULL && strcmp(*argv,"-p")==0) preserve++;
diff -rupN /home/pako/Downloads/lstrip/Makefile lua/lstrip/Makefile
--- /home/pako/Downloads/lstrip/Makefile	2014-07-03 16:27:01.000000000 +0200
+++ lua/lstrip/Makefile	2014-11-17 10:30:14.835273905 +0100
@@ -1,7 +1,7 @@
 # makefile for lstrip
 
 # change these to reflect your Lua installation (Lua 5.3!)
-LUA= /tmp/lhf/lua-5.3.0-work3
+LUA= $(LUA_DIR)
 LUAINC= $(LUA)/src
 LUALIB= $(LUA)/src
 LUASRC= $(LUA)/src
@@ -18,6 +18,7 @@ MYLIB= $(MYNAME)
 T= $(MYNAME)
 OBJS= $(MYNAME).o llex.o
 TEST= test.lua
+LUAEXECDIR = $(LUA_DIR)/../bin
 
 all:	test
 
@@ -26,6 +27,7 @@ test:	$T
 
 $T:	$(OBJS)
 	$(CC) -o $@ $(OBJS) $(LIBS)
+	cp $(MYNAME) $(LUAEXECDIR)
 
 llex.c:
 	sed '/void luaX_next/i#include "proxy.c"' $(LUASRC)/$@ > $@
@@ -33,6 +35,6 @@ llex.c:
 llex.o:	proxy.c
 
 clean:
-	-rm -f $(OBJS) core core.* a.out $(MYNAME)
+	-rm -f $(OBJS) core core.* a.out llex.c $(MYNAME)
 
 # eof
diff -rupN /home/pako/Downloads/lstrip/proxy.c lua/lstrip/proxy.c
--- /home/pako/Downloads/lstrip/proxy.c	2014-07-03 16:31:28.000000000 +0200
+++ lua/lstrip/proxy.c	2014-11-17 10:25:21.336521341 +0100
@@ -15,7 +15,7 @@ extern int lstrip_options;
 static void quote(const TString *ts)
 {
  const char* s=getstr(ts);
- size_t i,n=ts->tsv.len;
+ size_t i,n=ts->len;
  printf("%c",'"');
  for (i=0; i<n; i++)
  {
