diff -rupN /home/pako/programas/lua-5.3.0-beta/Makefile lua/lua-5.3.0/Makefile
--- /home/pako/programas/lua-5.3.0-beta/Makefile	2013-06-04 04:12:29.000000000 +0200
+++ lua/lua-5.3.0/Makefile	2014-11-17 10:25:21.336521341 +0100
@@ -4,13 +4,14 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================
 
 # Your platform. See PLATS for possible values.
-PLAT= none
+UNAME = `uname`
+PLAT= DetectOS
 
 # Where to install. The installation starts in the src and doc directories,
 # so take care if INSTALL_TOP is not an absolute path. See the local target.
 # You may want to make INSTALL_LMOD and INSTALL_CMOD consistent with
 # LUA_ROOT, LUA_LDIR, and LUA_CDIR in luaconf.h.
-INSTALL_TOP= /usr/local
+INSTALL_TOP= ../..
 INSTALL_BIN= $(INSTALL_TOP)/bin
 INSTALL_INC= $(INSTALL_TOP)/include
 INSTALL_LIB= $(INSTALL_TOP)/lib
@@ -36,7 +37,7 @@ RM= rm -f
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
 # Convenience platforms targets.
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= DetectOS Darwin Linux aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
 # What to install.
 TO_BIN= lua luac
diff -rupN /home/pako/programas/lua-5.3.0-beta/src/lua.c lua/lua-5.3.0/src/lua.c
--- /home/pako/programas/lua-5.3.0-beta/src/lua.c	2014-10-21 00:21:05.000000000 +0200
+++ lua/lua-5.3.0/src/lua.c	2014-11-17 10:53:30.656447843 +0100
@@ -567,6 +567,9 @@ static int pmain (lua_State *L) {
     lua_setfield(L, LUA_REGISTRYINDEX, "LUA_NOENV");
   }
   luaL_openlibs(L);  /* open standard libraries */
+#ifdef lua_userinit
+  lua_userinit(L);   /* APRIL: init packages */
+#endif
   createargtable(L, argv, argc, script);  /* create table 'arg' */
   if (!(args & has_E)) {  /* no option '-E'? */
     if (handle_luainit(L) != LUA_OK)  /* run LUA_INIT */
diff -rupN /home/pako/programas/lua-5.3.0-beta/src/luaconf.h lua/lua-5.3.0/src/luaconf.h
--- /home/pako/programas/lua-5.3.0-beta/src/luaconf.h	2014-10-21 16:38:46.000000000 +0200
+++ lua/lua-5.3.0/src/luaconf.h	2014-11-17 10:25:21.340521378 +0100
@@ -123,8 +123,8 @@
 ** hierarchy or if you want to install your libraries in
 ** non-conventional directories.
 */
-#define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR
 #if defined(_WIN32) 	/* { */
+#define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR
 /*
 ** In Windows, any exclamation mark ('!') in the path is replaced by the
 ** path of the directory of the executable file of the current process.
@@ -142,18 +142,43 @@
 		LUA_CDIR"..\\lib\\lua\\" LUA_VDIR "\\?.dll;" \
 		LUA_CDIR"loadall.dll;" ".\\?.dll"
 
-#else			/* }{ */
+#elif __APPLE__			/* }{ */
+
+#define LUA_VDIR	LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
+#define LUA_ROOT	"/usr/"
+#define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR
+#define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR
+#define LUA_PATH_DEFAULT				\
+  LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;"		\
+  LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" "./?.lua"
+#define LUA_CPATH_DEFAULT				\
+  LUA_CDIR"?.so;" LUA_CDIR"loadall.so;"			\
+  "./?.so;"
 
-#define LUA_ROOT	"/usr/local/"
-#define LUA_LDIR	LUA_ROOT "share/lua/" LUA_VDIR "/"
-#define LUA_CDIR	LUA_ROOT "lib/lua/" LUA_VDIR "/"
+#elif __linux  	         /* }{ */
+
+/* This defines DEB_HOST_MULTIARCH */
+#include "lua5.2-deb-multiarch.h"
+#define LUA_VDIR        LUA_VERSION_MAJOR "." LUA_VERSION_MINOR "/"
+#define LUA_ROOT        "/usr/local/"
+#define LUA_ROOT2       "/usr/"
+#define LUA_LDIR        LUA_ROOT "share/lua/" LUA_VDIR
+#define LUA_LDIR2       LUA_ROOT2 "share/lua/" LUA_VDIR
+#define LUA_CDIR        LUA_ROOT "lib/lua/" LUA_VDIR
+#define LUA_CDIR2       LUA_ROOT2 "lib/" DEB_HOST_MULTIARCH "/lua/" LUA_VDIR
+#define LUA_CDIR3       LUA_ROOT2 "lib/lua/" LUA_VDIR
 #define LUA_PATH_DEFAULT  \
-		LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
-		LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" \
-		"./?.lua;" "./?/init.lua"
+                LUA_LDIR"?.lua;"  LUA_LDIR"?/init.lua;" \
+                LUA_CDIR"?.lua;"  LUA_CDIR"?/init.lua;" "./?.lua;" \
+                LUA_LDIR2"?.lua;"  LUA_LDIR2"?/init.lua;" "./?.lua"
 #define LUA_CPATH_DEFAULT \
-		LUA_CDIR"?.so;" LUA_CDIR"loadall.so;" "./?.so"
-#endif			/* } */
+                LUA_CDIR"?.so;"  LUA_CDIR2"?.so;" \
+                LUA_CDIR3"?.so;"  LUA_CDIR"loadall.so;" "./?.so"
+#else			/* } */
+
+#error "Unknown platform"
+
+#endif
 
 
 /*
diff -rupN /home/pako/programas/lua-5.3.0-beta/src/Makefile lua/lua-5.3.0/src/Makefile
--- /home/pako/programas/lua-5.3.0-beta/src/Makefile	2014-10-16 02:09:23.000000000 +0200
+++ lua/lua-5.3.0/src/Makefile	2014-11-17 10:25:21.336521341 +0100
@@ -4,7 +4,8 @@
 # == CHANGE THE SETTINGS BELOW TO SUIT YOUR ENVIRONMENT =======================
 
 # Your platform. See PLATS for possible values.
-PLAT= none
+UNAME = `uname`
+PLAT= DetectOS
 
 CC= gcc
 CFLAGS= -O2 -Wall -Wextra -DLUA_COMPAT_5_2 $(SYSCFLAGS) $(MYCFLAGS)
@@ -26,7 +27,7 @@ MYOBJS=
 
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
-PLATS= aix ansi bsd freebsd generic linux macosx mingw posix solaris
+PLATS= DetectOS Darwin Linux aix ansi bsd freebsd generic linux macosx mingw posix solaris
 
 LUA_A=	liblua.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
@@ -47,6 +48,13 @@ ALL_T= $(LUA_A) $(LUA_T) $(LUAC_T)
 ALL_A= $(LUA_A)
 
 # Targets start here.
+DetectOS:
+	@make $(UNAME)
+
+Linux: linux
+
+Darwin: macosx
+
 default: $(PLAT)
 
 all:	$(ALL_T)
@@ -110,7 +118,7 @@ linux:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
 
 macosx:
-	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-lreadline" CC=cc
+	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_MACOSX" SYSLIBS="-ldl -lreadline" CC=cc
 
 mingw:
 	$(MAKE) "LUA_A=lua53.dll" "LUA_T=lua.exe" \
